### PR TITLE
Reorder Epoch Processing Functions

### PR DESCRIPTION
**Pull request by terencechain**
merge #2675 first

This PR reorders the epoch processing functions to read just like the spec, there's no new functionality being implemented

### CODE REVIEW

File: beacon-chain/core/state/transition_test.go [hunk 1]
```diff
@@ -394,251 +393,84 @@ func TestProcessBlock_PassesProcessingConditions(t *testing.T) {
...
-		LatestRandaoMixes:  randaoHashes,
-		LatestActiveIndexRoots: make([][]byte,
-			params.BeaconConfig().LatestActiveIndexRootsLength),
-		LatestSlashedBalances: make([]uint64,
-			params.BeaconConfig().LatestSlashedExitLength),
-		ValidatorRegistryUpdateEpoch: 0,
-		FinalizedEpoch:               1,
-	}
+func TestProcessEpoch_CantGetTgtAttsCurrEpoch(t *testing.T) {
+	e := params.BeaconConfig().SlotsPerEpoch
```
Comment (prestonvanloon) [inline beacon-chain/core/state/transition_test.go]: Why is this the case?

e = slots per epoch? So this is the 8th epoch? why?

Comment (terencechain) [reply beacon-chain/core/state/transition_test.go]: it's the 1st epoch right?

```go
&pb.BeaconState{Slot:e} // Slot: 64
```

Comment (prestonvanloon) [reply beacon-chain/core/state/transition_test.go]: can you name it `s` or `slot`? `e` has been shorthand for `epoch` elsewhere in the code

**Comment by nisdas** [comment]
lint fails

**Comment by terencechain** [comment]
> lint fails

fixed, thanks

**Comment by nisdas** [comment]
PR looks good by there are conflicts @terencechain

**Pull request merged**